### PR TITLE
dev/core#5241 - Don't drop tables during install

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Fixes https://lab.civicrm.org/dev/core/-/issues/5241

Before
----------------------------------------
The installer automatically drops existing tables. If accidentally run on a live site, all data would be destroyed. This is like having a loaded gun pointed at your database all the time (don't worry, as long as nothing accidentally triggers it, you'll be fine!).

After
----------------------------------------
Installer only creates tables. If they already exist, it will give an error, but no data will be destroyed.

Technical Details
---------
I don't know if there are any legit circumstances where we need this functionality. Are there?

### CODE REVIEW

**Comment by demeritcowboy** [comment]
I haven't tested but this seems ok since it should still be independent of `cv core:install` and the `-f` option and its checks, e.g. https://github.com/civicrm/cv/blob/28710d72f9e24150c473c4e6edfe8abefb6e17d8/src/Command/CoreInstallCommand.php#L151-L158. I.e. it shouldn't interfere with how it deals with tables.

**Comment by demeritcowboy** [comment]
I think you can mostly ignore my comment. When I run cv core:install it never hits this. I haven't really been keeping up with the codegen stuff so maybe this is used for something else? Which installer is this? It won't let me run either of the UI installers when it's already installed. Is there yet a 4th installer?

**Comment by totten** [comment]
tldr: I think this makes sense.

Yeah, there are some use-cases where one intentionally destroys the database (like Dave mentions `cv core:install -f`; then also `civibuild reinstall`; and I suppose the O.G. `setup.sh` workflow). But in all three cases, I believe other steps handle the actual destruction destruction (usu with `SHOW TABLES LIKE civicrm_ ... DROP TABLE $x` or `DROP DATABASE`).

Even if somebody *wants* to destroy the database, the `DROP TABLE` list in `civicrm.mysql` doesn't work particularly well -- which is why all those tools do it another way. (To show this, you have to step-back to the use-case: when you do a force-reinstall, it's often because you're switching to another version with a slightly different schema.... and you don't want to bother thinking about the details. The old schema and new schema are different. The new `DROP TABLE` list may not reflect everything in the DB -- leaving you with orphaned tables that cause trouble latter on.)

So I don't think any normal workflows rely on the `DROP TABLE` list in `civicrm.mysql` (or its live/logical successor). And if they do, then we should probably fix them...
